### DISCUSSION TITLE
[Bug]: Wrong launcher script detection, when a custom platform is used

### DISCUSSION THREAD

**Discussion by nosamad** [aspect-build/rules_js#2200] [General]
### What happened?

When using a custom target platform for a different os than the execution platform, the wrong launcher script is selected.
E.g. selecting a windows target platform on a linux execution platform, selects the windows launcher script and vice-versa.

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`:
aspect 5.2.1

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
0eca3b9

Language(s) and/or frameworks involved:
C/C++ and JS, TypeScript

### How to reproduce

```shell
Add a custom target platform for cross compile to a different OS than windows and try to build under windows.
```

### Any other information?
https://github.com/nosamad/rules_js/tree/topic/fix_launcher_script_selection

**Comment by gregmagolan** [comment]
I believe Bazel is doing the right thing here and building the js_binary for the target platform. For build actions that use a js_binary, those should still be built for the execution platform via the `cfg=exec` transition, which would be the host if you're not using RBE. What is your use case?

**Comment by nosamad** [comment]
> I believe Bazel is doing the right thing here and building the js_binary for the target platform. For build actions that use a js_binary, those should still be built for the execution platform via the `cfg=exec` transition, which would be the host if you're not using RBE. What is your use case?

My use case is, that i have a code-generator written in Javascript/ Typescript and a C/C++ Target Application, which depends on the generated sources and is compiled for the target platform.
When I set the target platform, to a non-windows target (e.g. GCC ARM Baremetal), the launcher script is wrong detected.
The host system is windows (not using RBE) and I get an error that the bash Script is not executable (Failed to start process).
The underlying toolchain (nodejs) is correctly selected for the execution platform - independent from the target platform.
I think the launcher script, should be selected according to the nodejs toolchain.

**Comment by gregmagolan** [comment]
Can you provide more details as to how the js_binary is being executed? Is it a custom rule? Are you calling `bazel build` or `bazel run`? Code snippets may be helpful.
